source/common/http/utility.cc

```diff
@@ -288,9 +288,15 @@ void Utility::sendLocalReply(bool is_grpc, StreamDecoderFilterCallbacks& callbac
   sendLocalReply(is_grpc,
                  [&](HeaderMapPtr&& headers, bool end_stream) -> void {
                    callbacks.encodeHeaders(std::move(headers), end_stream);
+                   if (end_stream) {
+                     callbacks.onEncodeComplete();
```

Comment:
If we move the functions to StreamDecoderFilter/StreamEncoderFilter, what's the best way to invoke them from Utility::sendLocalReply?

Reply:
Ultimately, send local reply goes back into the HCM and invokes filters, etc. We need to invoke the completion callbacks from within the HCM itself when a filter has decoded/encoded end_stream.

Reply:
Moved calls to the HCM (I think...I'm new to this project and learning as I go!).